### PR TITLE
#507 Commit characters other than enter/tab should not apply

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
- When in auto-complete mode make applicable span to be left part of the token to the current caret position.
- In explicit mode (`Ctrl+J`) use complete token for replace

Discussed this with @DinoV 
No entirely sure about naming, there is `ForceIntellisense` flag but looks that also `completeWord` term is used for the mode.

### CODE REVIEW

**Comment by huguesv** [comment]
If you haven't already, you should run the PythonToolsMockTests.EditorTests. If there's no regression, it probably means we need to add test(s) for this scenario.

FYI on my machine without your changes, they all pass except for `EnterCommitsCompleteNoNewLine`:
`Assert.Fail failed. Expected at least {__class__}, didn't find {__class__}. All: {}`

**Comment by MikhailArkhipov** [comment]
Will do

**Comment by MikhailArkhipov** [comment]
Pending merge of https://github.com/Microsoft/PTVS/issues/3398

**Comment by MikhailArkhipov** [comment]
`CompletionWithLongDocString` - passing on re-run. Might be that some random strings are not valid?
`BuiltinOverrideCompletions` - fails in master
`FromOSPathImportCompletions2x` - fails in master
`FromOSPathImportCompletions3x` - fails in master

File: Python/Product/PythonTools/PythonTools/Intellisense/IntellisenseController.cs [hunk 1]
```diff
@@ -876,22 +876,29 @@ private class ExpressionCompletionWalker : PythonWalker {
...
-            } else {
-                if (commitByDefault.HasValue) {
-                    foreach (var s in session.CompletionSets.OfType<FuzzyCompletionSet>()) {
-                        s.CommitByDefault = commitByDefault.GetValueOrDefault();
-                    }
+                return;
+            }
+
+            if (completeWord) {
+                session.SetCompleteWordMode();
```
Comment (zooba) [inline Python/Product/PythonTools/PythonTools/Intellisense/IntellisenseController.cs]: Don't we need to set this before the first round of completions are calculated? (When we chatted I suggested splitting the TriggerCompletion call into separate "create completion" and "start completion" calls, so we can set these properties in the middle.)

Comment (MikhailArkhipov) [reply Python/Product/PythonTools/PythonTools/Intellisense/IntellisenseController.cs]: Unfortunately, session is available only after `broker.TriggerCompletion()` returns. So I use both property and the `IntellisenseController.ForceCompletions` in the extension.

So we want to split Trigger into `ICompletionBroker.CreateCompletion` and the start right?

Comment (zooba) [reply Python/Product/PythonTools/PythonTools/Intellisense/IntellisenseController.cs]: Yeah. That's all that Trigger actually does, as well as handling failure to start safely. It's only 3-4 LoC.

Comment (MikhailArkhipov) [reply Python/Product/PythonTools/PythonTools/Intellisense/IntellisenseController.cs]: I've done that, pending running tests

**Pull request merged**